text/0502_aws-vpclattice.md

```diff
@@ -0,0 +1,506 @@
+   * @param domain
+   */
+  addCustomDomain(domain: string): void;
+
+  /**
+   * add a name for the service
+   * @default cloudformation will provide a name
+   */
+  addName(name: string): void;
+  /**
+   * grant access to the service
+   *
+   */
+  grantAccess(principals: iam.IPrincipal[]): void;
+  /**
+   * Apply the authAuthPolicy to the Service
+   */
+  applyAuthPolicyToService(): iam.PolicyDocument;
+  /**
+   * Add A policyStatement to the Auth Policy
+   */
+  addPolicyStatement(statement: iam.PolicyStatement): void;
+}
+```
+### Listener
+A listener is a process that checks for connection requests, using the protocol and port that you configure. The rules that you define for a listener determine how the service routes requests to its registered targets.
+
+It is not expected that a direct call to Listener will be made, instead the `.addListener()` should be used.
+
+
+``` typescript
+export interface IListener extends core.IResource {
+  /**
+  * The Amazon Resource Name (ARN) of the service.
+  */
+  readonly listenerArn: string;
+  /**
+  * The Id of the Service Network
+  */
+  readonly listenerId: string;
+
+  /**
+   * Add A Listener Rule to the Listener
+   */
+  addListenerRule(props: AddRuleProps): void;
+
+}
+```
+
+
+```typescript
+/**
+ * Propertys to Create a Lattice Listener
+ */
+export interface ListenerProps {
+  /**
+   *  * A default action that will be taken if no rules match.
+   *  @default 404 NOT Found
+  */
+  readonly defaultAction?: aws_vpclattice.CfnListener.DefaultActionProperty | undefined;
+  /**
+  * protocol that the listener will listen on
+  */
+  readonly protocol: Protocol
+  /**
+  * Optional port number for the listener. If not supplied, will default to 80 or 443, depending on the Protocol
+  * @default 80 or 443 depending on the Protocol
+
+  */
+  readonly port?: number | undefined
+  /**
+  * The Name of the service.
+  * @default CloudFormation provided name.
+  */
+  readonly name?: string;
+  /**
+   * The service
+   */
+  readonly serviceId: string;
+  /**
+   * the authpolicy for the service this listener is associated with
+   * @default none.
+   */
+  readonly serviceAuthPolicy?: iam.PolicyDocument | undefined
+}
+```
+
+### TargetGroups
+
+A VPC Lattice target group is a collection of targets, or compute resources, that run your application or service. Targets can be EC2 instances, IP addresses, Lambda functions, Application Load Balancers, or Kubernetes Pods. 
+
+```typescript
+ * Create a vpc lattice TargetGroup.
+ * Implemented by `TargetGroup`.
+ */
+export interface ITargetGroup extends core.IResource {
+  /**
+   * The id of the target group
+   */
+  readonly targetGroupId: string
+  /**
+   * The Arn of the target group
+   */
+  readonly targetGroupArn: string;
+
+}
+```
+
+
+
+```typescript
+/**
+ * Properties for a Target Group, Only supply one of instancetargets, lambdaTargets, albTargets, ipTargets
+ */
+export interface TargetGroupProps {
```

Comment:
This needs a tighter contract. We shouldn't have a prop that is dependent on the others. Instead, I would create a prop called `targets: Target[]` that works similar to Schedule (linked above) with Static functions for the various target types. The input of each function except for the Lambda Targets function would also take in a `config` prop.

Reply:
Agreed and implemented